src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -74,6 +93,184 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+                    value: {
+                        currency_code: 'USD',
+                        value: (parseFloat(String(baseOrderAmount))).toFixed(2),
+                        breakdown: {
+                            item_total: {
+                                currency_code: 'USD',
+                                value: baseOrderAmount,
+                            },
+                        },
+                    },
+                },
+            ]);
+        }
+
+        if (shippingRequired && availableShippingOptions?.length === 0) {
+            return actions.reject();
+        }
+
+        const shippingOptions = availableShippingOptions?.map((option: AvaliableShippingOption) => {
+            let isSelected = false;
+            // Buyer has chosen shipping option on PP list and address the same
+            if (this._isAddressSame(this._currentShippingAddress, this._submittedShippingAddress)
+            && option.id === data.selected_shipping_option.id
+            ) {
+            shippingAmount = data.selected_shipping_option.amount.value;
+            isSelected = true;
+            } else if (option.isRecommended) {
+            shippingAmount = parseFloat(String(option.cost)).toFixed(2);
+            isSelected = true;
+            }
+
+            return {
+                id: option.id,
+                type: 'SHIPPING',
+                label: option.description,
+                selected: isSelected,
+                amount: {
+                    value: parseFloat(String(option.cost)).toFixed(2),
+                    currency_code: 'USD',
+                },
+            };
+        });
+        this._submittedShippingAddress = this._currentShippingAddress;
+
+        if (shippingOptions) {
+            // @ts-ignore
+            this._selectedShippingOptionId = shippingOptions.find(({ selected }) => selected).id;
+            actions.order.patch([
+                {
+                    op: 'replace',
+                    path: '/purchase_units/@reference_id==\'default\'/amount',
+                    value: {
+                        currency_code: 'USD',
+                        value: (parseFloat(String(baseOrderAmount)) + parseFloat(shippingAmount)).toFixed(2),
+                        breakdown: {
+                            item_total: {
+                                currency_code: 'USD',
+                                value: baseOrderAmount,
+                            },
+                            shipping: {
+                                currency_code: 'USD',
+                                value: shippingAmount,
+                            },
+                        },
+                    },
+                },
+                {
+                    op: 'add',
+                    path: '/purchase_units/@reference_id==\'default\'/shipping/options',
+                    value: shippingOptions,
+                },
+            ]);
+        }
+
+        return actions.resolve();
```

Comment:
Can we return the result of the patch instead of making actions resolve in the end?

```suggestion
            this._selectedShippingOptionId = shippingOptions.find(({ selected }) => selected).id;
            
            return await actions.order.patch([
                {
                    op: 'replace',
                    path: '/purchase_units/@reference_id==\'default\'/amount',
                    value: {
                        currency_code: 'USD',
                        value: (parseFloat(String(baseOrderAmount)) + parseFloat(shippingAmount)).toFixed(2),
                        breakdown: {
                            item_total: {
                                currency_code: 'USD',
                                value: baseOrderAmount,
                            },
                            shipping: {
                                currency_code: 'USD',
                                value: shippingAmount,
                            },
                        },
                    },
                },
                {
                    op: 'add',
                    path: '/purchase_units/@reference_id==\'default\'/shipping/options',
                    value: shippingOptions,
                },
            ]);
        }
```